src/Kiota.Builder/Refiners/TypeScriptRefiner.cs

```diff
@@ -19,6 +21,12 @@ public override Task Refine(CodeNamespace generatedCode, CancellationToken cance
             cancellationToken.ThrowIfCancellationRequested();
             DeduplicateErrorMappings(generatedCode);
             RemoveMethodByKind(generatedCode, CodeMethodKind.RawUrlConstructor, CodeMethodKind.RawUrlBuilder);
+            ConvertUnionTypesToWrapper(
```

Comment:
there's something I need more context on here. Why are we converting the union types to wrappers if we have writers dedicated to union types?

Reply:
I don't have a compelling reason for this, but there are some useful processes that occur within the `ConvertUnionTypesToWrapper` function. For instance, method names are refined and significant tasks like adding discriminator properties are performed. If I skip this step, I will need to find an alternative method to achieve the same results while being consistent with other languages, which will likely take more time.

Reply:
ok in the interest of time I'm ok with sacrificing a little runtime performance and maintainability so we get this out of the door. I'd like @andrueastman 's opinion on the matter in order to keep me in check.

Reply:
@baywet I think we can keep this one as is for now. 
What we should probably do is add a comment here to explain that we are calling this to "temporarily" achieve a more consistent codeDom to other languages, but the later refiner stages will either way move out the added discriminator methods to files and the writers will not write out the wrapper type(but its constituent types) as they are not referenced directly later on (it would need to be converted to an interface..).